daml-lf/encoder/src/test/scala/com/digitalasset/daml/lf/archive/testing/DamlLfEncoderTest.scala

```diff
@@ -67,12 +69,26 @@ class DamlLfEncoderTest
 
   }
 
-  private def getModules(dar: Dar[(PackageId, DamlLf.ArchivePayload)]) =
+  private val preInternalizationVersions = List.range(0, 7).map(_.toString).toSet
+
+  private def getModules(dar: Dar[(PackageId, DamlLf.ArchivePayload)]) = {
     for {
       pkgWithId <- dar.main +: dar.dependencies
       (_, pkg) = pkgWithId
-      segments <- pkg.getDamlLf1.getModulesList.asScala.map(_.getName.getSegmentsList.asScala)
+      version = pkg.getMinor
+      internedStrings = pkg.getDamlLf1.getInternedStringsList.asScala.toArray
+      dottedNames = pkg.getDamlLf1.getInternedDottedNamesList.asScala.map(
+        _.getSegmentsInternedStrList.asScala.map(internedStrings(_))
+      )
+      segments <- pkg.getDamlLf1.getModulesList.asScala.map(
+        mod =>
+          if (preInternalizationVersions(version))
```

Comment:
Would it make sense to check the major version as well here? I'm not familiar with the LF versioning scheme hence this may be an irrelevant question. If we indeed need to check the major version I'd suggest to change the naming to avoid confusion (e.g., `preInternalizationVersions` -> `preInternalizationMinorVersions`).